lib/env-bundler.js

```diff
@@ -177,7 +177,16 @@ const componentCodeExtractors = {
 
 const componentResponseProcessor = {
   page: async (response, token) => {
-    const pages = await Promise.all(response.data.payload.map(async (page) => {
+    let pages = response.data.payload;
```

Comment:
Mutating the same variable in a lot of places, also using the same variable that you are mapping through.

```
    // extract homepage
    const res_homepage = await axios.get(apiUrlTable.pageDetails.replace('{code}', 'homepage'), {
      headers: { Authorization: `Bearer ${token}` },
    });
    const pages = [res_homepage.data.payload, ...response.data.payload];

    const fetchedPages = await Promise.all(pages.map(async (page) => {
    // etc
```